compiler/damlc/daml-stdlib-src/DA/Internal/Interface.daml

```diff
@@ -0,0 +1,49 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | MOVE Prelude interface functionality
+module DA.Internal.Interface (
+  HasInterfaceTypeRep(..),
+  HasToInterface(..),
+  HasFromInterface(..),
+  Implements,
+  toInterface,
+  toInterfaceContractId,
+  fromInterfaceContractId,
+) where
+
+import DA.Internal.Prelude
+import DA.Internal.Template.Functions
+import DA.Internal.LF
+
+class HasInterfaceTypeRep i where
+  interfaceTypeRep : i -> TypeRep
+
+class HasToInterface t i where
+  toInterface_ : t -> i
+
+toInterface : forall i t. HasToInterface t i => t -> i
+toInterface = toInterface_
+
+class HasFromInterface t i where
+  fromInterface : i -> Optional t
+
+type Implements t i =
+  ( HasInterfaceTypeRep i
```

Comment:
> Doesn't this cause issues with functions that use constraints like `(Implements t Token) => ...`?

I also thought it would, but those errors get silenced by `MonoLocalBinds`, which is part of the default language extensions in Daml. 

> I feel like we can leave `HasinterfaceTypeRep i` off of the `Implements t i` constraint tuple

We could, but I don't think it's necessary given the above

Reply:
Ah, nice! 👍